src/code/z_actor.c

```diff
@@ -457,439 +1254,3581 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+    actorCtx->flags.chest = cycleFlags[0];
+    actorCtx->flags.swch[0] = cycleFlags[1];
+    actorCtx->flags.swch[1] = cycleFlags[2];
+    if (globalCtx->sceneNum == SCENE_INISIE_R) {
+        cycleFlags = gSaveContext.cycleSceneFlags[globalCtx->sceneNum];
+    }
+    actorCtx->flags.collectible[0] = cycleFlags[4];
+    actorCtx->flags.clearedRoom = cycleFlags[3];
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BE5CC.s")
+    TitleCard_ContextInit(gameState, &actorCtx->titleCtxt);
+    func_800B6468(globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BE63C.s")
+    actorCtx->absoluteSpace = NULL;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BE680.s")
+    Actor_SpawnEntry(actorCtx, actorEntry, gameState);
+    Actor_TargetContextInit(&actorCtx->targetContext, actorCtx->actorList[ACTORCAT_PLAYER].first, globalCtx);
+    func_800B9120(actorCtx);
+    Fault_AddClient(&D_801ED8A0, (void*)Actor_PrintLists, actorCtx, NULL);
+    func_800B722C(gameState, (Player*)actorCtx->actorList[ACTORCAT_PLAYER].first);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BF7CC.s")
+/**
+ * Spawns the actors in the current setup (of the current scene/setup/room triple)
+ * Only spawns actors based on the time flags embedded in their rotation parameters
+ */
+void Actor_SpawnSetupActors(GlobalContext* globalCtx, ActorContext* actorCtx) {
+    if (globalCtx->numSetupActors > 0) {
+        ActorEntry* actorEntry = globalCtx->setupActorList;
+        s32 temp_fp = actorCtx->unkC;
+        s32 temp_s1;
+        s32 phi_v0;
+        s32 i;
+
+        func_800B9120(actorCtx);
+        func_800BA8B8(globalCtx, &globalCtx->actorCtx);
+
+        temp_s1 = (actorCtx->unkC * 2) & 0x2FF;
+
+        for (i = 0; i < globalCtx->numSetupActors; i++) {
+            phi_v0 = ((actorEntry->rot.x & 7) << 7) | (actorEntry->rot.z & 0x7F);
+            if (phi_v0 == 0) {
+                phi_v0 = 0x3FF;
+            }
+
+            if (!(phi_v0 & temp_fp) && (phi_v0 & actorCtx->unkC) &&
+                (!(gSaveContext.eventInf[1] & 0x80) || !(phi_v0 & temp_s1) || !(actorEntry->id & 0x800))) {
+                Actor_SpawnEntry(&globalCtx->actorCtx, actorEntry, &globalCtx->state);
+            }
+            actorEntry++;
+        }
+
+        // Prevents re-spawning the setup actors
+        globalCtx->numSetupActors = -globalCtx->numSetupActors;
+    }
+}
+
+typedef struct {
+    /* 0x00 */ GlobalContext* globalCtx;
+    /* 0x04 */ Actor* actor;
+    /* 0x08 */ u32 updateActorIfSet;
+    /* 0x0C */ u32 unkC;
+    /* 0x10 */ Actor* unk10;
+    /* 0x14 */ Player* player;
+    /* 0x18 */ u32 runMainIfSet; // Bitmask of actor flags. The actor will only have main called if it has at least 1
+                                 // flag set that matches this bitmask
+} UpdateActor_Params;            // size = 0x1C
+
+#ifdef NON_MATCHING
+Actor* Actor_UpdateActor(UpdateActor_Params* params) {
```

Comment:
Exactly.
Also I don't think this struct should be place in a header, because it is used only by `Actor_UpdateActor`, which is a function not intended to be used outside of this system